pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/TableViews.java

```diff
@@ -0,0 +1,173 @@
+      return responseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
+          "{\"error\" : \"Table name and view type are required\"}");
+    }
+    return getTableState(tableName, view, tableType);
+  }
+
+  public static class TableView
+  {
+    @JsonProperty("OFFLINE")
+    Map<String, Map<String, String>> offline;
+    @JsonProperty("REALTIME")
+    Map<String, Map<String, String>> realtime;
+  }
+
+  // we use name "view" to closely match underlying names and to not
+  // confuse with table state of enable/disable
+  @HttpVerb("get")
+  @Summary("Get table idealstate or external view")
+  @Tags({"table"})
+  @Paths({"/tables/{tableName}/{view}"})
+  private Representation getTableState(
+      @Parameter(name = "tableName", in = "path", description = "Table name(without type)", required = true)
+      String tableName,
+      @Parameter(name="view", in = "path", description = "Table idealstate or external view", required = true)
+      String view,
+      @Parameter(name = "tableType", in="query", description = "Table type", required = false)
+      TableType tableType) {
+    TableView tableView;
+    if (view.equalsIgnoreCase("idealstate")) {
+      tableView = getTableIdealState(tableName, tableType);
+    } else if (view.equalsIgnoreCase("externalview")) {
+      tableView = getTableExternalView(tableName, tableType);
+    } else {
+      return responseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
+          "{\"error\" : \"Bad view name: " + view + ". Expected idealstate or externalview\" }");
+    }
+
+    if (tableView.offline == null && tableView.realtime == null) {
+      return responseRepresentation(Status.CLIENT_ERROR_NOT_FOUND,
+          "{\"error\" : \"Table not found}");
+    }
+
+    ObjectMapper mapper = new ObjectMapper();
+    try {
+      String response = mapper.writeValueAsString(tableView);
+      return responseRepresentation(Status.SUCCESS_OK, response);
+    } catch (JsonProcessingException e) {
+      LOGGER.error("Failed to serialize {} for table {}", tableName, view, e);
+      return responseRepresentation(Status.SERVER_ERROR_INTERNAL, "{\"error\": \"Error serializing response\"}");
+    }
+  }
+
+  private TableView getTableExternalView(String tableNameOptType, TableType tableType) {
+    TableView tableView = new TableView();
+    if (tableType == null || tableType == TableType.OFFLINE) {
+      tableView.offline = getExternalView(tableNameOptType, TableType.OFFLINE);
+    } else if (tableType == null || tableType == TableType.REALTIME) {
+      tableView.realtime = getExternalView(tableNameOptType, TableType.REALTIME);
+    }
+    return tableView;
+  }
+
+  private TableView getTableIdealState(String tableNameOptType, TableType tableType) {
+    TableView tableView = new TableView();
+    if (tableType == null || tableType == TableType.OFFLINE) {
+      tableView.offline = getIdealState(tableNameOptType, TableType.OFFLINE);
+    }
+    if (tableType == null || tableType == TableType.REALTIME) {
```

Comment:
(tableType == null) means that you do realtime and offline. In case of external view you only get offline but in the case of idealstate you get both.